packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -718,6 +725,10 @@ class _AppBarState extends State<AppBar> {
     final AppBarTheme appBarTheme = AppBarTheme.of(context);
     final ScaffoldState? scaffold = Scaffold.maybeOf(context);
     final ModalRoute<dynamic>? parentRoute = ModalRoute.of(context);
+    final FlexibleSpaceBarSettings? settings = context.dependOnInheritedWidgetOfExactType<FlexibleSpaceBarSettings>();
+    final Set<MaterialState> states = <MaterialState>{
+      if (settings?.isScrolledUnder == true) MaterialState.scrolledUnder,
```

Comment:
So the plain `AppBar` does not have a way to set this yet right? Would this be handled in a follow-up PR? Is it feasible

Reply:
It should be possible to _just_ implement this support in AppBar, by having the AppBar ask its Scaffold for ScrollNotifications instead of passing the isScrolledUnder information from SliverAppBar via FlexibleSpaceBarSettings. Will give that a try.

Reply:
I've updated the PR to include support for AppBar.